mergify_engine/clients/http.py

```diff
@@ -64,12 +64,16 @@ class HTTPClientSideError(httpx.HTTPStatusError):
     def message(self) -> str:
         # TODO(sileht): do something with errors and documentation_url when present
         # https://developer.github.com/v3/#client-errors
-        to_cast = (
-            self.response.json()["errors"][0]["message"]
-            if "errors" in self.response.json()
-            else self.response.json()["message"]
-        )
-        return typing.cast(str, to_cast)
+        response = self.response.json()
+        message = response["message"] if "message" in response else "No message found"
```

Comment:
I wasn't sure about setting the message to `None`or to `No message found`, in case there is no `message`key in the payload, although this case is practically impossible, WDYT ?

Reply:
I'd prefer to change the typing and return `None` if no message is found.
But then it's up to the caller to handle the `None` case.